src/GoogleAnalyticsTracker.js

```diff
@@ -1,103 +1,53 @@
+   * @param trackerId {String}
+   * @param customDimensionsFieldsIndexMap {{fieldName: fieldIndex}} Custom dimensions field/index pairs
+   */
+  constructor(trackerId, customDimensionsFieldsIndexMap) {
+    this.id = trackerId;
+    this.customDimensionsFieldsIndexMap = customDimensionsFieldsIndexMap;
+  }
+
+  /**
+   * If Tracker has customDimensionsFieldsIndexMap, it will transform
+   * customDimensions map pairs {field: value} to {fieldIndex: value}.
+   * Otherwise customDimensions are passed trough untouched.
+   * Underlay native methods will transform provided customDimensions map to expected format.
+   * Google analytics expect dimensions to be tracker with 'dimension{index}' keys,
+   * not dimension field names.
+   * @param customDimensions {Object}
+   * @returns {Object}
+   */
+  transformCustomDimensionsFieldsToIndexes(customDimensions) {
+    if (this.customDimensionsFieldsIndexMap) {
+      const mappedCustomDimensions = {};
+      Object.keys(this.customDimensionsFieldsIndexMap).forEach(key => {
+        const dimensionIndex = this.customDimensionsFieldsIndexMap[key];
+        const customDimensionVal = customDimensions[key];
+        const customDimensionValType = typeof customDimensionVal;
+        // Custom dimensions accept only strings and numbers
+        if (customDimensionValType === 'string' || customDimensionValType === 'number') {
+          mappedCustomDimensions[dimensionIndex] = customDimensions[key];
+        }
+      });
+      return mappedCustomDimensions;
     }
-
-    this.addTrackerId = this.addTrackerId.bind(this);
-  }
-
-  /**
-   * Backwards compatibility, old API support.
-   * @param {String} trackerId
-   */
-  setTrackerId(trackerId) {
-    this.setTrackerIds([trackerId]);
-  }
-
-  /**
-   * @param {Array} trackerIds
-   */
-  setTrackerIds(trackerIds) {
-    this.trackerIds = new Set(trackerIds);
-  }
-
-  /**
-   * Add trackerId to trackers set.
-   * @param {String} trackerId
-   */
-  addTrackerId(trackerId) {
-    this.trackerIds.add(trackerId);
-  }
-
-  /**
-   * Add trackerIds to trackers set.
-   * @param {Array} trackerId
-   */
-  addTrackerIds(trackerIds) {
-    trackerIds.forEach(this.addTrackerId);
-  }
-
-  /**
-   * Remove trackerId from tracker ids set.
-   * @param {String} trackerId
-   */
-  removeTrackerId(trackerId) {
-    this.trackerIds.delete(trackerId);
-  }
-
-  /**
-   * Call provided method with provided arguments for every trackerId.
-   * Injects trackerId as first argument.
-   *
-   * @param {Function} method
-   * @param {Arguments} args Passed method arguments
-   */
-  callNativeGoogleAnalyticsMethod(method, args) {
-    this.trackerIds.forEach(
-      trackerId =>
-        method.call(GoogleAnalyticsBridge, trackerId, ...args)
-    );
+    return customDimensions;
   }
 
   /**
    * Track the current screen/view
    * @param  {String} screenName The name of the current screen
    */
   trackScreenView(screenName) {
```

Comment:
I want to keep signature visible, that is why I kept arguments, bot sure if IDEs will suggest it only based on JSDoc.